src/main/kotlin/com/exactpro/th2/kafka/client/RawMessageProcessor.kt

```diff
@@ -0,0 +1,99 @@
+package com.exactpro.th2.kafka.client
+
+import com.exactpro.th2.common.grpc.RawMessage
+import com.exactpro.th2.common.grpc.RawMessageBatch
+import com.exactpro.th2.common.message.toTimestamp
+import mu.KotlinLogging
+import java.time.Instant
+import java.util.concurrent.Executors
+import java.util.concurrent.Future
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.BlockingQueue
+import java.util.concurrent.LinkedBlockingQueue
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.locks.ReentrantLock
+import kotlin.concurrent.thread
+import kotlin.concurrent.withLock
+
+class RawMessageProcessor(
```

Comment:
Why don't use this one instead https://github.com/th2-net/th2-common-utils-j/blob/master/src/main/kotlin/com/exactpro/th2/common/utils/message/MessageBatcher.kt?

Reply:
looks like it doesn't ensures the correct ordering of messages.